plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getVoidLogger } from '@backstage/backend-common';
+import { ClusterDetails, KubernetesClustersSupplier } from '../types/types';
+import { KubernetesProxy, KubernetesProxyServices } from './KubernetesProxy';
+
+import { Request } from 'express';
+
+import 'buffer';
+
+jest.mock('node-fetch');
```

Comment:
a hint here is to search through the repo for occurrences of the `msw` library being used in backend packages - it's an excellent request capturing and response shaping tool for tests, making sure that you don't have to mock out fetch but instead interacting with an actual "real fake" server

Reply:
Removed the mock, using MSW